egs/wsj/s5/steps/libs/nnet3/xconfig/self_layer.py

```diff
@@ -0,0 +1,218 @@
+# Copyright 2016    Johns Hopkins University (Author: Daniel Povey)
+#           2016    Vimal Manohar
+#           2018    Tom Ko, Zhu Yingke
+# Apache 2.0.
+
+""" This module contains the statistics extraction and self attention layer.
+"""
+
+from __future__ import print_function
+import re
+from libs.nnet3.xconfig.basic_layers import XconfigLayerBase
+
+
+class XconfigSelfLayer(XconfigLayerBase):
+    """This class is for parsing lines like
+    self-layer name=attention config=mean+stddev(-99:3:9:99) input=tdnn1
```

Comment:
Better to use a different name like weighted-stats-layer instead of a generic one like self-layer.